server/src/main/java/com/linecorp/centraldogma/server/CommitRetentionConfig.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.centraldogma.server;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+
+import javax.annotation.Nullable;
+
+import com.cronutils.model.Cron;
+import com.cronutils.model.CronType;
+import com.cronutils.model.definition.CronDefinitionBuilder;
+import com.cronutils.parser.CronParser;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import com.linecorp.centraldogma.server.storage.repository.Repository;
+
+/**
+ * A configuration for retaining commits in a {@link Repository}.
+ */
+public final class CommitRetentionConfig {
+
+    // The minimum of minRetentionCommits
+    private static final int MINIMUM_MINIMUM_RETENTION_COMMITS = 1000;
```

Comment:
1,000 seems too small as the default min value. How about 5,000? Some mega repos in LINE are larger than 250,000 commits but most repo are less than 2000 commits. I guess 5,000 would keep all histories of normal repos and will not trigger creating a secondary repo.